gateway/middleware.go

```diff
@@ -482,8 +482,7 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 
 					if greaterThanFloat64(policy.Rate, ar.Limit.Rate) {
 						ar.Limit.Rate = policy.Rate
-						//if policy.Partitions.RateLimit then we must set this value in the global data of the key
-						if greaterThanFloat64(policy.Rate, session.Rate) || policy.Partitions.RateLimit {
+						if greaterThanFloat64(policy.Rate, session.Rate) {
```

Comment:
@buger would we not want to calculate the actual rate/per ratio?

Reply:
Even we apply another logic here, this fix is needed because multiple policies may have rate limit partition enabled and it doesn’t respect the logic.

Reply:
Ah, it got merged automatically when I approved it. Which probably fix the current bug. 
But I think @titpetric has a point here, in a sense that we treat `RateLimit` and `Rate` field separately here, but they actually highly connected. So the proper logic here should be actually:
```go
if (policy.Rate/policy.Per) > (ar.Limit.Rate/ar.Limit.Per) {
   ar.Limit.Rate = policy.Rate
   ar.Limit.Per = policy.Per
   session.Rate = policy.Rate
   session.Per = policy.Per
}
```